### ISSUE TITLE
task failure not handled by celery when broker connection error

### ISSUE THREAD

**Issue by tisdall** [celery/celery#3241]
In my case I'm using RabbitMQ as a broker.  If I try calling `.delay` on a task when RabbitMQ is down, I get a socket exception because a connection is attempted and it fails.  It seems like in other failure cases celery has machinery for handling exceptions (such as a task running and throwing an exception), but in this case an exception bubbles up out of celery.  So, it seems like I need to wrap every call in a try-except if I want to handle connection errors.  Shouldn't this be something celery does?

**Comment by tisdall** [comment]
For example, shouldn't a call to `.apply_async(retry=False)` simply fail at sending the task and return without an exception?

**Comment by ask** [comment]
The retry will also reraise the exception when giving up, and I'm not sure if silently eating errors for tasks would be a good idea?

**Comment by tisdall** [comment]
True.  What about registering an error handler with celery that's called when a task is tried but the connection to RabbitMQ fails?  That way you only need to register the handler once and won't need to wrap every call in a try-except.  The default handler could be that it outputs the error to STDERR or whatever the logging facility is.  (On the off chance someone is relying on socket exceptions, I suppose you could turn this functionality on/off with a config option)
